hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -672,6 +672,7 @@ protected void getAllObjectInternal(List<Data> keys, List<Object> resultingKeyVa
         if (keys.isEmpty()) {
             return;
         }
+        long time = System.currentTimeMillis();
```

Comment:
@fredang Good job, mate! Started to look into this myself, but you beat me to it! 

The only thing I did differently was that I have inserted the statement recording the start time after the creating the operationFactory. This makes it more consistent to how the respective existing code works for the put operation. 
eg

```
            OperationFactory operationFactory = operationProvider.createGetAllOperationFactory(name, keys);
            final long time = System.currentTimeMillis();
            responses = operationService.invokeOnPartitions(SERVICE_NAME, operationFactory, partitions);
```

It could be debated that your position better because it also takes into account the factory creation. Going down this road would mean that to keep the two behaviours consistent it would also need to be changed for putAll. But this would mean two changes instead of one. 

Best,
Giorgos